### PR TITLE
An 2455/intermediate swaps raydium

### PR DESCRIPTION
**Pull request by tarikceric**
Create a separate model for intermediate swaps performed on Raydium

### CODE REVIEW

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/swaps/silver__swaps_intermediate_raydium.sql [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+
+    SELECT
+        *
+    FROM
+        {{ ref('silver__events') }}
+    WHERE
+        program_id IN (
+            --raydium program_ids
+            '675kPX9MHTjS2zt1qfr1NYHuzeLXfQM9H24wFSUt1Mp8',
+            '5quBtoiQqxF9Jv6KYKctB59NT3gtJD2Y65kdnB1Uev3h',
```
Comment (desmond-hui) [inline models/silver/swaps/silver__swaps_intermediate_raydium.sql:18]: It's probably worth adding these other two 

```
export const LIQUIDITY_POOL_PROGRAM_ID_V2 = 'RVKd61ztZW9GUwhRbbLoYVRE5Xf1B2tVscKqwZqXgEr'
export const LIQUIDITY_POOL_PROGRAM_ID_V3 = '27haf8L6oxUeXrHrgEgsexjSY5hbVUWEmvv9Nyxg8vQv'
```

resource https://github.com/raydium-io/raydium-ui/blob/f1e0d0ab5b1f7ccb90e84e5a81cd58d54b69ac3c/src/utils/ids.ts

File: models/silver/swaps/silver__swaps_intermediate_raydium.sql [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+            '93BgeoLHo5AdNbpqy9bD12dtfxtA5M2fh3rj72bE35Y3',
```
Comment (desmond-hui) [inline models/silver/swaps/silver__swaps_intermediate_raydium.sql:19]: Possibly remove `93BgeoLHo5AdNbpqy9bD12dtfxtA5M2fh3rj72bE35Y3`? Might not be a direct swap...needs further investigation

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/swaps/silver__swaps_intermediate_raydium.sql [hunk 3]
```diff
@@ -0,0 +1,396 @@
+{{ config(
+    materialized = 'incremental',
+    unique_key = ["block_id","tx_id","swap_index"],
+    merge_predicates = ["DBT_INTERNAL_DEST.block_timestamp::date >= LEAST(current_date-7,(select min(block_timestamp)::date from {{ this }}__dbt_tmp))"],
+    cluster_by = ['block_timestamp::DATE','_inserted_timestamp::DATE'],
+) }}
+
+WITH base_events AS(
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/swaps/silver__swaps_intermediate_raydium.sql:8]: New filters added

File: models/silver/swaps/silver__swaps_intermediate_raydium.sql (hunk 1 above)
Comment (tarikceric) [reply models/silver/swaps/silver__swaps_intermediate_raydium.sql:18]: Txs with these program_ids are all LP actions and not swaps, so they were not added

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**